mongo_service.go

```diff
@@ -46,14 +47,41 @@ func (service *MongoService) closeSession() {
 	service.session.Close()
 }
 
-func (service *MongoService) isCurrentNodeMaster() bool {
+func (service *MongoService) isNodeForBackup() bool {
 	db := service.session.DB(defaultDb)
 	result := make(map[string]interface{})
 	err := db.Run(bson.M{"isMaster": 1}, result)
 	if err != nil {
 		log.Panic("Can't check if node is master, db.isMaster() fails", err.Error(), err)
 	}
-	return result["ismaster"].(bool)
+
+	if result["me"] == nil || result["hosts"] == nil {
+		warn.Println("Node is not part of a cluster.")
+		return false
+	}
+
+	master := result["ismaster"].(bool)
+	if master {
+		info.Println("This node is master.")
+		return false
+	}
+
+	thisNode := result["me"].(string)
+	primary := result["primary"].(string)
+	allNodes := result["hosts"].([]string)
+	sort.Strings(allNodes)
+	var lowestSecondary string
+	if allNodes[0] == primary {
+		lowestSecondary = allNodes[1]
+	} else {
+		lowestSecondary = allNodes[0]
+	}
```

Comment:
So we're sorting the host:port strings returned in the 'hosts' field, and the highest on that list that isn't master is the node to back up on? Is this for consistency, so that backups are usually performed on the same node?